src/controllers/LiveDataController.ts

```diff
@@ -24,8 +24,14 @@ export async function init() {
     stsApi.onDidLiveProcessDisconnect(resetProcessInfo);
     stsApi.onDidLiveProcessUpdate(updateProcessInfo);
     // proposed API on vscode-spring-boot extension. check before calling.
-    stsApi.onDidLiveProcessGcPausesMetricsUpdate?.(updateProcessGcPausesMetrics);
-    stsApi.onDidLiveProcessMemoryMetricsUpdate?.(updateProcessMemoryMetrics);
+    if(typeof stsApi.onDidLiveProcessGcPausesMetricsUpdate === "function") {
```

Comment:
previous statement is sufficient enough for me, you make it safer by checking whether the api is a function.